definitionsconsistencycheck.cmake

```diff
@@ -1,11 +1,11 @@
 get_directory_property( DirDefs COMPILE_DEFINITIONS )
 
 # Reset the definition file
-file(WRITE cmake.definitions "")
+file(WRITE $ENV{__IntermediatesDir}/cmake.definitions "")
```

Comment:
Alright. And for the problem of Windows build you've mentioned with #4675, would it be ok to make the script ignore file-not-found and smoothly move on afterwards?